docs/appendices/release-notes/unreleased.rst

```diff
@@ -168,3 +168,6 @@ Fixes
 - Fixed an issue that leads to stuck write queries if the
   :ref:`indices.breaker.query.limit` is set to ``-1`` as the value was
   interpreted as a byte value instead of disabling any breaking.
+
+- Fixed an issue that caused ``col IS NULL``/``col IS NOT NULL`` to
+  match/don't match empty objects or empty arrays of objects.
```

Comment:
not sure about `NOT` - it's a reversed case, maybe can drop the mention completely?